boards/arc/hsdk/pinmux.c

```diff
@@ -12,11 +12,10 @@ static int board_pinmux_init(const struct device *device)
 {
 	ARG_UNUSED(device);
 
-	const struct device *pinmux = device_get_binding(CONFIG_PINMUX_NAME);
+	const struct device *pinmux = DEVICE_DT_GET(DT_NODELABEL(pinctrl));
+
+	__ASSERT_NO_MSG(device_is_ready(pinmux));
```

Comment:
Are we going to make assert in this case policy?  This driver is one of the few ones that wouldn't fault if `pinmux` was null.  Is changing that appropriate?

Reply:
is it possible that pinmux is ever NULL now?  I thought because of how DEVICE_DT_GET works we'd get a compile error, thus i removed the NULL check.

Reply:
No, `pinmux` won't be null, but the previous implementation would cause board init to return an error (which would be ignored) and the app would keep running.  Now it will either blow up on the assert, or proceed to use the uninitialized device, depending on whether asserts are enforced.

Preserving existing behavior would mean not asserting, but instead replacing `pinmux ==NULL` with `device_is_ready(pinmux)` and retaining the error return.

IOW this code was correct in testing the driver before using it, but your mod changes behavior that I'm not sure needed to be changed.  The other conversions blithely used the device whether it was valid or not, so had undefined behavior and you defined the behavior to be assert, which I think is justifiable.

Does that make sense?

Reply:
I don't believe the error return was intentional. I'm ok with an assert, the pinmux should always be there on this board - not like we have versions with and without.

Reply:
OK, with maintainer approval I'm okay with the change.